src/main/kotlin/org/elm/ide/inspections/ElmUnresolvedReferenceInspection.kt

```diff
@@ -47,14 +35,11 @@ class ElmUnresolvedReferenceAnnotator : Annotator {
             refs.removeIf { it is QualifiedReference }
         }
 
-        // Give each handler a chance to deal with the unresolved ref before falling back on an error
-        outerLoop@
         for (ref in refs.filter { it.resolve() == null }) {
-            for (handler in handlers) {
-                if (handler(ref, element, holder)) {
-                    continue@outerLoop
-                }
-            }
+            // Give each handler a chance to deal with the unresolved ref before falling back on an error
+            if (handleTypeAnnotation(ref, element, holder)) continue
```

Comment:
this is a lot better. I think I had been in Spring-land too long when I thought of the handler thing 😄